### ISSUE TITLE
Installation problems with python 3.11

### ISSUE THREAD

**Issue by martinthomson** [ietf-tools/svgcheck#23]
@chris-wood seems to be having trouble installing this tool and I can't see where the error originates.  Details in https://github.com/martinthomson/i-d-template/issues/358

Python 3.11 removed the "U" mode qualifier, which is probably what is going on here.  However, I can't see where the code that uses that flag lives.  It doesn't look like it is in this module.

**Comment by rjsparks** [comment]
The metadata for setup needs to be modified to better reflect what versions of python have been _tested_.

**Comment by chris-wood** [comment]
@rjsparks is this something y'all can fix? It makes using the toolchain sort of painful right now.

**Comment by rjsparks** [comment]
We will work towards making it work on 3.11, but it won't be soon. Do you have the option of using an earlier python in the meantime?

**Comment by chris-wood** [comment]
I just work around it by removing the svgcheck requirement, which is a fine temporary workaround.

**Comment by nedmsmith** [comment]
I tried building with 3.10.9 and it didn't work

**Comment by rjsparks** [comment]
To be clear, right now the supported and tested versions are what you see here:
https://github.com/ietf-tools/svgcheck/blob/main/tox.ini

This line tells the story best: `envlist = py{27,36,37,38,39}-{linux,macos,windows}`

We'll work to make what shows in setup.py more accurate, and to extend what is supported/tested to more recent pythons.

**Comment by dkg** [comment]
> @chris-wood seems to be having trouble installing this tool and I can't see where the error originates. Details in [martinthomson/i-d-template#358](https://github.com/martinthomson/i-d-template/issues/358)
> 
> Python 3.11 removed the "U" mode qualifier, which is probably what is going on here. However, I can't see where the code that uses that flag lives. It doesn't look like it is in this module.

fwiw, in commit 04dbd28ef6825951023a5d17d8b3c1436acc04d9 the `U` modifier was removed; this is after the release of 0.6.0, though, so if @chriss-wood is trying to use it from PyPI (which only has 0.6.0) it would still be an issue for him (see #1).

**Comment by nedmsmith** [comment]
> Python 3.11 removed the "U" mode qualifier, which is probably what is going on here.

The output below seems to be expecting mode='rU' but found mode='rUb'. I have no idea what might be causing the difference or if it is benign. 

I tried running against Python3.9, but still have the same issue. Possibly, my experience is not related to https://github.com/martinthomson/i-d-template/issues/358 ?

```
Collecting svgcheck
  Using cached svgcheck-0.6.0.tar.gz (18 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [11 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/private/var/folders/bv/wdn1dcmj5xg9wg8nrnpxnmpw0000gp/T/pip-install-j1g5mnpm/svgcheck_c18337739ff44883b04c2eebe3b36eb8/setup.py", line 65, in <module>
          """ + "\n".join([ changelog_entry_template % entry for entry in parse("changelog")[:3] ])
                                                                          ^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/bv/wdn1dcmj5xg9wg8nrnpxnmpw0000gp/T/pip-install-j1g5mnpm/svgcheck_c18337739ff44883b04c2eebe3b36eb8/setup.py", line 32, in parse
          changelog = open(changelog, mode='rU', encoding='utf-8')
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "<frozen codecs>", line 906, in open
      ValueError: invalid mode: 'rUb'
      [end of output]
```

**Comment by kesara** [comment]
Version 0.6.1 was not published on PyPI.
I have published [0.6.1](https://github.com/ietf-tools/svgcheck/releases/tag/v0.6.1) and [0.6.2](https://github.com/ietf-tools/svgcheck/releases/tag/v0.6.2) to PyPI.
This should fix the installation issues up to Python 3.10.

Installing under Python 3.11 still fails because [rfctools_common](https://github.com/ietf-tools/RfcEditor/tree/master/rfctools_common)
I think trying to make use of `xml2rfc` instead of `rfctools_common` might be a better step forward.

**Comment by kesara** [comment]
[svgcheck 0.7.0](https://github.com/ietf-tools/svgcheck/releases/tag/v0.7.0) removes dependency on `rfctools_common` this fixes the installation issue under Python 3.11.
